src/Sylius/Component/Core/OrderProcessing/StateResolver.php

```diff
@@ -61,35 +79,48 @@ public function resolvePaymentState(OrderInterface $order)
      */
     public function resolveShippingState(OrderInterface $order)
     {
-        $order->setShippingState($this->getShippingState($order));
-    }
+        if (OrderShippingStates::SHIPPED === $order->getShippingState()) {
+            return;
+        }
+        /** @var Collection $shipments */
+        $shipments = $order->getShipments();
+        $shipmentCount = $shipments->count();
+        $shipmentInShippedStateAmount = $shipments->filter(
+            function (ShipmentInterface $shipment) {
+                return ShipmentInterface::STATE_SHIPPED === $shipment->getState();
+            })
+            ->count()
+        ;
```

Comment:
Anyway, we can extract getting amount of shipment of given type to some private function as it is repeated three times.